diff --git a/src/gromacs/ewald/pme_gpu_grid.cu b/src/gromacs/ewald/pme_gpu_grid.cu
index 5393aa7..b06f1c8 100644
--- a/src/gromacs/ewald/pme_gpu_grid.cu
+++ b/src/gromacs/ewald/pme_gpu_grid.cu
@@ -801,9 +801,11 @@ static void receiveAndSend(DeviceBuffer<float> sendBuf,
                            MPI_Comm            comm)
 {
     // send data to dest rank and recv from src rank
+#if GMX_MPI
     MPI_Irecv(recvBuf, recvCount, MPI_FLOAT, src, tag, comm, recvRequest);
 
     MPI_Isend(sendBuf, sendCount, MPI_FLOAT, dest, tag, comm, sendRequest);
+#endif
 }
 
 void pmeGpuGridHaloExchange(const PmeGpu* pmeGpu, gmx_wallcycle* wcycle)
diff --git a/src/gromacs/mdlib/mdgraph_gpu_impl.cu b/src/gromacs/mdlib/mdgraph_gpu_impl.cu
index 8d5d0cf..99d54cd 100644
--- a/src/gromacs/mdlib/mdgraph_gpu_impl.cu
+++ b/src/gromacs/mdlib/mdgraph_gpu_impl.cu
@@ -73,9 +73,11 @@ MdGpuGraph::Impl::Impl(const DeviceStreamManager& deviceStreamManager,
 
     if (havePPDomainDecomposition_)
     {
+#    if GMX_MPI
         MPI_Barrier(mpiComm_);
         MPI_Comm_size(mpiComm_, &ppSize_);
         MPI_Comm_rank(mpiComm_, &ppRank_);
+#    endif
     }
 }
 
@@ -106,6 +108,7 @@ void MdGpuGraph::Impl::enqueueEventFromAllPpRanksToRank0Stream(GpuEventSynchroni
     {
         if (ppRank_ == remotePpRank)
         {
+#    if GMX_MPI
             // send event to rank 0
             MPI_Send(&event,
                      sizeof(GpuEventSynchronizer*), //NOLINT(bugprone-sizeof-expression)
@@ -113,11 +116,13 @@ void MdGpuGraph::Impl::enqueueEventFromAllPpRanksToRank0Stream(GpuEventSynchroni
                      0,
                      0,
                      mpiComm_);
+#    endif
         }
         else if (ppRank_ == 0)
         {
             // rank 0 enqueues recieved event
             GpuEventSynchronizer* eventToEnqueue;
+#    if GMX_MPI
             MPI_Recv(&eventToEnqueue,
                      sizeof(GpuEventSynchronizer*), //NOLINT(bugprone-sizeof-expression)
                      MPI_BYTE,
@@ -125,6 +130,7 @@ void MdGpuGraph::Impl::enqueueEventFromAllPpRanksToRank0Stream(GpuEventSynchroni
                      0,
                      mpiComm_,
                      MPI_STATUS_IGNORE);
+#    endif
             eventToEnqueue->enqueueWaitEvent(stream);
         }
     }
@@ -140,8 +146,10 @@ void MdGpuGraph::Impl::enqueueRank0EventToAllPpStreams(GpuEventSynchronizer* eve
 {
     if (havePPDomainDecomposition_)
     {
+#    if GMX_MPI
         // NOLINTNEXTLINE(bugprone-sizeof-expression)
         MPI_Bcast(&event, sizeof(GpuEventSynchronizer*), MPI_BYTE, 0, mpiComm_);
+#    endif
     }
     event->enqueueWaitEvent(stream);
 }
@@ -160,12 +168,14 @@ void MdGpuGraph::Impl::disableForDomainIfAnyPpRankHasCpuForces(bool disableGraph
     if (havePPDomainDecomposition_)
     {
         // If disabled on any domain, disable on all domains
+#    if GMX_MPI
         MPI_Allreduce(&disableGraphAcrossAllPpRanks,
                       &disableGraphAcrossAllPpRanks_,
                       sizeof(bool),
                       MPI_BYTE,
                       MPI_SUM,
                       mpiComm_);
+#    endif
     }
 }
 
@@ -194,7 +204,9 @@ void MdGpuGraph::Impl::startRecord(GpuEventSynchronizer* xReadyOnDeviceEvent)
     wallcycle_sub_start(wcycle_, WallCycleSubCounter::MdGpuGraphWaitBeforeCapture);
     if (havePPDomainDecomposition_)
     {
+#    if GMX_MPI
         MPI_Barrier(mpiComm_);
+#    endif
     }
     wallcycle_sub_stop(wcycle_, WallCycleSubCounter::MdGpuGraphWaitBeforeCapture);
 
@@ -233,7 +245,9 @@ void MdGpuGraph::Impl::startRecord(GpuEventSynchronizer* xReadyOnDeviceEvent)
         enqueueRank0EventToAllPpStreams(
                 helperEvent_.get(), deviceStreamManager_.stream(gmx::DeviceStreamType::NonBondedLocal));
         // The synchronization below should not be needed, see #4674
+#    if GMX_MPI
         MPI_Barrier(mpiComm_);
+#    endif
 
         // Fork NB non-local stream from NB local stream on each rank
         helperEvent_->markEvent(deviceStreamManager_.stream(gmx::DeviceStreamType::NonBondedLocal));
@@ -320,7 +334,9 @@ void MdGpuGraph::Impl::endRecord()
     // Sync all tasks before closing timing region, since the graph capture should be treated as a collective operation for timing purposes.
     if (havePPDomainDecomposition_)
     {
+#    if GMX_MPI
         MPI_Barrier(mpiComm_);
+#    endif
     }
     wallcycle_sub_stop(wcycle_, WallCycleSubCounter::MdGpuGraphCapture);
     wallcycle_stop(wcycle_, WallCycleCounter::MdGpuGraph);
@@ -374,7 +390,9 @@ void MdGpuGraph::Impl::createExecutableGraph(bool forceGraphReinstantiation)
     // Sync all tasks before closing timing region, since the graph instantiate or update should be treated as a collective operation for timing purposes.
     if (havePPDomainDecomposition_)
     {
+#    if GMX_MPI
         MPI_Barrier(mpiComm_);
+#    endif
     }
     wallcycle_sub_stop(wcycle_, WallCycleSubCounter::MdGpuGraphInstantiateOrUpdate);
     wallcycle_stop(wcycle_, WallCycleCounter::MdGpuGraph);
@@ -391,7 +409,9 @@ void MdGpuGraph::Impl::launchGraphMdStep(GpuEventSynchronizer* xUpdatedOnDeviceE
     wallcycle_sub_start(wcycle_, WallCycleSubCounter::MdGpuGraphWaitBeforeLaunch);
     if (havePPDomainDecomposition_)
     {
+#    if GMX_MPI
         MPI_Barrier(mpiComm_);
+#    endif
     }
     wallcycle_sub_stop(wcycle_, WallCycleSubCounter::MdGpuGraphWaitBeforeLaunch);
 
